dvc/stage.py

```diff
@@ -982,22 +983,30 @@ def checkout(
         relink=False,
         filter_info=None,
     ):
+        def _fspath_dir(path):
+            """Return string with "/" at the end if it's a dir."""
+            p = os.path.join(path, "") if os.path.isdir(path) else fspath(path)
+            return relpath(p, self.repo.root_dir)
+
         checkouts = {"failed": [], "added": [], "modified": []}
         for out in self._filter_outs(filter_info):
             try:
-                is_modified = out.checkout(
+                result = out.checkout(
                     force=force,
                     tag=self.tag,
                     progress_callback=progress_callback,
                     relink=relink,
                     filter_info=filter_info,
                 )
-                if is_modified:
-                    checkouts["modified"].append(out.path_info)
-                elif is_modified is False:
-                    checkouts["added"].append(out.path_info)
+                added, modified = result or (None, None)
+                if added:
+                    checkouts["added"].append(_fspath_dir(out.path_info))
+                elif modified:
+                    checkouts["modified"].append(_fspath_dir(out.path_info))
             except CheckoutError as exc:
-                checkouts["failed"].extend(exc.target_infos)
+                checkouts["failed"].extend(
+                    [target for target in _fspath_dir(exc.target_infos)]
```

Comment:
is it misprint?
```python
[_fspath_dir(target) for target in exc.target_infos]
```

Reply:
Nice catch! Thanks. I should write more tests. :)